### ISSUE TITLE
Downloads since last build counter

### ISSUE THREAD

**Issue by rnkn** [melpa/melpa#1698]
Just for the narcissists among us, I'd like to put out there the possibility of adding a download counter that resets upon each package build, effectively giving (approximately) the number of active users.

This would be in addition to the total downloads counter.

Thoughts?

**Comment by snowman** [comment]
it would great if such statistic are available, so that i could try the most popular packages first, find out which become trending in short-time(means the curve is steep), get the most real data without search webpage :<

**Comment by purcell** [comment]
We actually have all the data for this. The full download histories for every version of every package are in a ~2GB sqlite DB on the build server. But efficiently exporting useful slices of that data that are not prohibitively large is the tricky bit, and there's a bit of development effort involved for which I don't think there has been a sufficiently compelling use case so far.

**Comment by purcell** [comment]
Like, I wonder if we could run some OSS reporting/remixing software (metabase or similar?) and expose it that way.

**Comment by rnkn** [comment]
@purcell if I understand correctly, at present the total download value is just a sum of IP requests per package name as parsed in the logs? If so, could this same parser not do a two-pass count, first counting the total of `PACKAGE-*`, and then counting just `PACKAGE-VERSION`?

**Comment by purcell** [comment]
> if I understand correctly, at present the total download value is just a sum of IP requests per package name as parsed in the logs?


@rnkn Not quite. Here's the corresponding logging and query code: https://github.com/melpa/melpa/blob/06cf1a337bf4bc355c3b297f6387e658f668d3b1/docker/logprocessor/process_log.py#L145-L148


> If so, could this same parser not do a two-pass count, first counting the total of `PACKAGE-*`, and then counting just `PACKAGE-VERSION`?

Potentially, but you'd need to select only the latest version within each group for each package, otherwise you'd end up with all the per-version-per-package stats and that would be enormously larger.
